// react-switch-selector/dist/index.modern.js
import e,{createElement as o,useState as t,useEffect as r}from"react";import{setup as n,styled as i}from"goober";import{shouldForwardProp as l}from"goober/should-forward-prop";function d(){return(d=Object.assign||function(e){for(var o=1;o<arguments.length;o++){var t=arguments[o];for(var r in t)Object.prototype.hasOwnProperty.call(t,r)&&(e[r]=t[r])}return e}).apply(this,arguments)}let a,s,c,p,u=e=>e;const b=["fontSize","fontColor","selectedFontColor","optionsAmount","optionBorderRadius","isRawText","selectedIndex","border","backgroundColor","selectedBackgroundColor","wrapperBorderRadius","selectionIndicatorMargin","disabled"];n(o,void 0,void 0,l(e=>!b.includes(e)));const g=i("div")(a||(a=u`
    display: flex;
    border-radius: ${0};
    border: ${0};
    background: ${0};
    width: 100%;
    height: 100%;
    position: relative;
    opacity: ${0};
    overflow: hidden;

    &::before {
        top: 50%;
        left: ${0}%;
        content: "";
        position: absolute;
        height: calc(100% - ${0}px);
        width: calc(
            ${0}%${0}${0}px
        );
        border-radius: ${0};
        border: ${0}px solid
            ${0};
        background: ${0};
        transition: left 0.1s linear, background 0.1s linear;
        transform: translateY(-50%);
        z-index: 1;
        box-sizing: content-box;
    }
`),({wrapperBorderRadius:e})=>"number"==typeof e?`${e}px`:e,e=>e.border,e=>e.backgroundColor,e=>e.disabled?.7:1,e=>e.selectedIndex/e.optionsAmount*100,e=>2*e.selectionIndicatorMargin,e=>1/e.optionsAmount*100," - ",e=>2*e.selectionIndicatorMargin,({optionBorderRadius:e})=>"number"==typeof e?`${e}px`:e,e=>e.selectionIndicatorMargin,e=>e.backgroundColor,e=>e.selectedBackgroundColor),f=i("div")(s||(s=u`
    display: flex;
    align-items: center;
    height: 100%;
    width: ${0}%;
    border-radius: ${0};
`),e=>1/e.optionsAmount*100,({optionBorderRadius:e})=>"number"==typeof e?`${e}px`:e),h=i("label")(c||(c=u`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    z-index: 2;
    transition: color 0.1s linear;
    cursor: ${0};
    font-size: ${0};
    color: ${0};
`),e=>e.disabled?"not-allowed":"pointer",e=>e.isRawText?e.fontSize+"px":"unset",e=>e.isRawText?e.selected?e.selectedFontColor:e.fontColor:"unset"),C=i("input")(p||(p=u`
    width: 0;
    height: 0;
    opacity: 0;
    z-index: -1;
    position: absolute;
    pointer-events: none;
`)),m={backgroundColor:"#ecf0f1",selectedBackgroundColor:"#2ecc71",fontColor:"#000",selectedFontColor:"#fff"};export default o=>{var n;const{onChange:i=(()=>{}),options:l=[],initialSelectedIndex:a=0}=o,s=!!l[a],[c,p]=t(s?a:0),{border:u=0,backgroundColor:b=m.backgroundColor,selectedBackgroundColor:x=m.selectedBackgroundColor,wrapperBorderRadius:w=20,optionBorderRadius:$=18,fontSize:k=14,fontColor:B=m.fontColor,selectedFontColor:y=m.selectedFontColor,selectionIndicatorMargin:v=2,forcedSelectedIndex:R,disabled:I=!1,name:z}=o;return r(()=>{void 0!==R&&l[R]&&R!==c&&p(R)},[R,l,c]),l.length?e.createElement(g,{selectedIndex:c,optionsAmount:l.length,className:"react-switch-selector-wrapper "+(I?"react-switch-selector-disabled":""),border:u,backgroundColor:b,selectedBackgroundColor:(null==(n=l[c])?void 0:n.selectedBackgroundColor)||x,wrapperBorderRadius:w,optionBorderRadius:$,selectionIndicatorMargin:v,disabled:I,role:"radiogroup","aria-labelledby":z},l.map((o,t)=>{const r=c===t,n=`${null!=z?z:"rss"}-option-${t}`,a="string"==typeof o.label;return e.createElement(f,{key:n,optionsAmount:l.length,className:"react-switch-selector-option react-switch-selector-option-"+(r?"selected":"unselected"),optionBorderRadius:$},e.createElement(h,d({className:"react-switch-selector-option-label",selected:r,isRawText:a,disabled:I,"aria-disabled":I,htmlFor:n},a?{fontSize:k,fontColor:o.fontColor||B,selectedFontColor:o.selectedFontColor||y}:{}),e.createElement(C,{type:"radio",id:n,name:z,onChange:()=>(e=>{I||e===c||(p(e),i(l[e].value))})(t),checked:r,"aria-checked":r,tabIndex:r?0:-1}),o.label))})):null};
